.github/workflows/markdown-linter.yml

```diff
@@ -0,0 +1,49 @@
+name: '[CI/CD] Markdown linter'
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '**.md'
+permissions:
+  contents: read
+jobs:
+  markdown-linter:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install mardownlint
+        run: npm install -g markdownlint-cli@0.33.0
+      - name: Checkout project
+        uses: actions/checkout@v3
+      - name: Execute markdownlint
+        run: |
+          # Using the Github API to detect the files changed as git merge-base stops working when the branch is behind
+          # and jitterbit/get-changed-files does not support pull_request_target
+          URL="https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }}/files"
+          files_changed_data=$(curl -s --header 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' -X GET -G "$URL")
+          files_changed="$(echo "$files_changed_data" | jq -r '.[] | .filename')"
+          md_files="$(echo "$files_changed" | grep -o ".*\.md$" | sort | uniq || true)"
+          # Create an empty file, useful when the PR changes ignored files
+          touch ${{runner.temp}}/output
+          exit_code=0
+          markdownlint -o ${{runner.temp}}/output ${md_files[@]} || exit_code=$?
```

Comment:
Why is there an array expansion in `${md_files[@]}`?

Would it be a problem if we have any path with whitespaces? I guess we don't have to care about that case anyways.

Reply:
The first try was using `'**/*.md'` but it took so long. Users mainly touch one or two files so I decided to calculate and pass them to the linter.

About the whitespaces, we might run into troubles if we had them